overlord/servicestate/service_control.go

```diff
@@ -55,6 +55,10 @@ type ServiceAction struct {
 	// inactive and not disabled services in snap-name, and also svc1 regardless
 	// of the state svc1 is in.
 	ExplicitServices []string `json:"explicit-services,omitempty"`
+	// RestartOnlyActive is only for "restart" and "reload-or-restart"
+	// actions, and when set it restarts only the active services, that is,
+	// enabled non-running services are ignored.
+	RestartOnlyActive bool `json:"restart-only-active,omitempty"`
```

Comment:
the default of this is wrong for tasks that were made by old snapd though?

Reply:
Right, but I'm not sure on the concern here, tasks from older changes would have been already run by the old snapd, and anyway we are changing behavior, it does probably not matter if it affects a task created before snapd refreshes?

Reply:
independent changes might be running while we update snapd, and it matters again if the tasks were about quota groups, I fear we need to make the default be the old behavior and deal with that

Reply:
I've reverted the flag now